test/unittests/evm_test.cpp

```diff
@@ -946,18 +946,22 @@ TEST_F(evm, shift_overflow)
 
 TEST_F(evm, undefined_instructions)
 {
+    auto latest_metrics = evmc_get_instruction_metrics_table(EVMC_MAX_REVISION);
+
     for (auto i = 0; i <= EVMC_MAX_REVISION; ++i)
     {
-        auto r = evmc_revision(i);
-        auto names = evmc_get_instruction_names_table(r);
+        rev = evmc_revision(i);
+        auto names = evmc_get_instruction_names_table(rev);
 
         for (uint8_t opcode = 0; opcode <= 0xfe; ++opcode)
         {
             if (names[opcode] != nullptr)
                 continue;
 
-            auto res = vm.execute(*this, r, {}, &opcode, sizeof(opcode));
-            EXPECT_EQ(res.status_code, EVMC_UNDEFINED_INSTRUCTION) << hex(opcode);
+            auto code = latest_metrics[opcode].num_stack_arguments * push(0) + evmc_opcode(opcode);
+
+            execute(30000000, code);
```

Comment:
I haven't dig into this too much yet, but it seemed that with default gas (which is `std::numeric_limits<int64_t>::max()` analyze phase has some kind of integer overflow (because for undefined opcodes gas price is defined as `-1`), then execution end with `EVMC_OUT_OF_GAS` instead of `EVMC_UNDEFINED_INSTRUCTION`

Reply:
That should be noticeable if you build with UBSan, `-DSANITIZE=undefined`.

Reply:
Yeah it reports it
```
/home/andrei/dev/evmone/lib/evmone/instructions.cpp:1164:25: runtime error: signed integer overflow: 9223372036854775807 - -1 cannot be represented in type 'long int'
```